### PR TITLE
platform: Remove weak tag from important variables

### PR DESCRIPTION
**Pull request by d3m3vilurr**
all platform should define this.

Related #95

### CODE REVIEW

File: optional_platform.cpp [hunk 1]
```diff
@@ -13,10 +13,6 @@ __attribute__((weak)) int logMessage(log_priority priority, const char *fmt, ...
 
 __attribute__((weak)) extern const bool HAS_HW_KEY2 = false;
```
Comment (angelsl) [inline optional_platform.cpp]: I think we should remove this too.

Comment (d3m3vilurr) [reply optional_platform.cpp]: IMO, this field is useless. current time this section only can enter `CAN_RESET` is true.
and these platforms can have the BF2 keys.

If you agree, I'll remove related codes.

Comment (kitling) [reply optional_platform.cpp]: I'd argue that powersaves can probably reset the cart, but more than likely does not handle key2 on it own, which means that it'd have to be done in software. Was the main reason I argued for having a way to do software key2.

Comment (angelsl) [reply optional_platform.cpp]: (n.b. by remove I mean this weak definition)

**Pull request merged**